src/RcppInterface.cpp

```diff
@@ -3,15 +3,74 @@
 #include <R.h>
 #include "Distributions.cpp"
 
+// The initialization of the static variables was originally planned to be inline.
+// See https://github.com/diego-urgell/BinSeg/releases/tag/std_rcpp17 for more information about this.
+// This solution is temporary, and will be changed as soon as the external bug is fixed.
+
 std::vector<std::string> mean_norm::param_names = {"before_mean", "after_mean"};
 std::vector<std::string> var_norm::param_names = {"before_var", "after_var"};
 std::vector<std::string> meanvar_norm::param_names = {"before_mean", "after_mean", " before_var", "after_var"};
 std::vector<std::string> negbin::param_names = {"before_prob", "after_prob"};
 std::vector<std::string> poisson::param_names = {"before_rate", "after_rate"};
 std::vector<std::string> exponential::param_names = {"before_rate", "after_rate"};
 
+std::string mean_norm::factoryName = "mean_norm";
+std::string var_norm::factoryName = "var_norm";
+std::string meanvar_norm::factoryName = "meanvar_norm";
+std::string negbin::factoryName = "negbin";
+std::string poisson::factoryName = "poisson";
+std::string exponential::factoryName = "exponential";
```

Comment:
@diego-urgell please change the DISTRIBUTION macro so that it sets this there, so you do not have to declare these names in a separate place.
same for description below.

Reply:
Yes, it was that way before, we used the double expansion trick for this. But it is not possible now since it relies on the inline keyword :(